pkg/storage/ingest/reader.go

```diff
@@ -61,16 +59,17 @@ type PartitionReader struct {
 	reg    prometheus.Registerer
 }
 
-func NewPartitionReaderForPusher(kafkaCfg KafkaConfig, partitionID int32, pusher Pusher, logger log.Logger, reg prometheus.Registerer) (*PartitionReader, error) {
+func NewPartitionReaderForPusher(kafkaCfg KafkaConfig, partitionID int32, consumerGroup string, pusher Pusher, logger log.Logger, reg prometheus.Registerer) (*PartitionReader, error) {
 	consumer := newPusherConsumer(pusher, reg, logger)
-	return newPartitionReader(kafkaCfg, partitionID, consumer, logger, reg)
+	return newPartitionReader(kafkaCfg, partitionID, consumerGroup, consumer, logger, reg)
```

Comment:
do you want to fold the ConsumerGroup into KafkaConfig in the interest of fewer parameters? (I'd even say partitionID should have been there in the first place, but maybe later on)

Reply:
Right now, `KafkaConfig` is what you user can configure while input args is internally config. I would keep this distinction for now. I find it easier to reason about.